content/en/docs/components/pipelines/sdk/build-pipeline.md

```diff
@@ -0,0 +1,465 @@
+returned as files.
+
+In your component's specification, you define the components inputs and outputs
+and how the inputs and output paths are passed to your program as command-line
+arguments. You can pass small inputs, such as short strings or numbers, to your
+component by value. Large inputs, such as datasets, must be passed to your
+component as file paths. Outputs are written to the paths that Kubeflow
+Pipelines provides.
+
+Python function-based components make it easier to build pipeline components
+by building the component specification for you. Python function-based
+components also handle the complexity of passing inputs into your component
+and passing your function’s outputs back to your pipeline.
+
+Learn more about how [Python function-based components handle inputs and
+outputs][python-function-component-data-passing]. 
+
+## Getting started building a pipeline
+
+The following sections demonstrate how to get started building a Kubeflow
+pipeline by walking through the process of converting a Python script into
+a pipeline.
+
+### Design your pipeline
+
+The following steps walk through some of the design decisions you may face
+when designing a pipeline.
+
+1.  Evaluate the process. In the following example, a Python function downloads
+    a zipped tar file (`.tar.gz`) that contains several CSV files, from a
+    public website. The function extracts the CSV files and then merges them
+    into a single file.
+
+[container-op]: https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ContainerOp
+[component-spec]: https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/
+[python-function-component]: https://www.kubeflow.org/docs/components/pipelines/sdk/python-function-components/
+[component-dev]: https://www.kubeflow.org/docs/components/pipelines/sdk/component-development/
+[python-function-component-data-passing]: https://www.kubeflow.org/docs/components/pipelines/sdk/python-function-components/#understanding-how-data-is-passed-between-components
+[prebuilt-components]: https://www.kubeflow.org/docs/examples/shared-resources/
+
+
+```python
+import glob
+import pandas as pd
+import tarfile
+import urllib.request
+    
+def download_and_merge_csv(url: str, output_csv: str):
+  with urllib.request.urlopen(url) as res:
+    tarfile.open(fileobj=res, mode="r|gz").extractall('data')
+  df = pd.concat(
+      [pd.read_csv(csv_file, header=None) 
+       for csv_file in glob.glob('data/*.csv')])
+  df.to_csv(output_csv, index=False, header=False)
+```
+
+2.  Run the following Python command to test the function. 
+
+
+```python
+download_and_merge_csv(
+    url='https://storage.googleapis.com/ml-pipeline-playground/iris-csv-files.tar.gz', 
+    output_csv='merged_data.csv')
+```
+
+3.  Run the following to print the first few rows of the
+    merged CSV file.
+
+
+```python
+$ head merged_data.csv
+```
+
+4.  Design your pipeline. For example, consider the following pipeline designs.
```

Comment:
"consider the following pipeline designs:"

Reply:
Thanks! I have another PR coming up with some small changes to the doc, so I will include this change in that PR.